hook.php

```diff
@@ -531,27 +496,16 @@ function plugin_formcreator_hook_purge_ticket(CommonDBTM $item) {
 
    $id = $item->getID();
 
-   // Update status of form answer, if any
+   // Update the formanswer's status (for cases where a form answer has several tickets)
    $formAnswer = new PluginFormcreatorFormAnswer();
-   $formAnswer->getFromDBByCrit([
-      'id' => new QuerySubQuery([
-         'SELECT' => 'items_id',
-         'FROM'   => Item_Ticket::getTable(),
-         'WHERE'  => [
-            'itemtype' => PluginFormcreatorFormAnswer::getType(),
-            'tickets_id' => $id,
-         ]
-      ])
-   ]);
-   if (!$formAnswer->isNewItem()) {
+   if ($formAnswer->getFromDbByTicket($id)) {
```

Comment:
if inverted

if a form answer is associated to the ticket being purged then the status if this fom answer must e updated.